### ISSUE TITLE
Release `0.3`

### ISSUE THREAD

**Issue by CraftSpider** [CraftSpider/department#2]
Currently, release 0.3 is waiting on a couple things, hopefully:

- [ ] `JustMetadata<T>`: Allows implementing CoerceUnsized on most handle types - https://github.com/rust-lang/rust/pull/97052
- [ ] GAT Stabilization: Allows using GATs on stable - https://github.com/rust-lang/rust/pull/96709
- [ ] Optionally, something to allow `Handle<T, for<T> This<T> = Self::Handle<T>>` to be added as a bound to handles

**Comment by CraftSpider** [comment]
Closing in favor of project
